### DISCUSSION TITLE
Building a universal library for macOS

### DISCUSSION THREAD

**Discussion by fdidron** [libusb/hidapi#473] [Q&A] [answered]
Hello,

I'm currently trying to build a universal version of hidapi. I've generated an Xcode project using make, and am now in the process to get that setup.

I will post my progress here, but in the meantime if anyone has any pointers on how to achieve that would be great :)

**Comment by fdidron** [comment, marked as answer]
Well I found a way to do it with cmake right after posting this, maybe this can be helpful.

Add the `CMAKE_OSX_ARCHITECTURES=x86_64;arm64` to the cmake command.

Following the [Cmake building guide](https://github.com/libusb/hidapi/blob/master/BUILD.cmake.md):

```sh
# Executed from the root of the repository
mkdir build
cmake .. -DCMAKE_OSX_ARCHITECTURES=x86_64;arm64
cmake --build .
cmake --build . --target install
```

To make sure the result is a universal lib, run `file /usr/local/lib/libhidapi.dylib` you should see

``` 
Mach-O universal binary with 2 architectures:
[x86_64:Mach-O 64-bit dynamically linked shared library x86_64]
[arm64:Mach-O 64-bit dynamically linked shared library arm64]
 ```


Edited to add @Youw suggestion.

**Reply by Youw** [reply]
Don't modify CMake files, pass it as a command-line parameter: `cmake .. -DCMAKE_OSX_ARCHITECTURES=x86_64;arm64`

**Reply by fdidron** [reply]
@Youw Thanks for the suggestion, I edited the solution.
